test/test-compare-statements.js

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env node
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+
+const compareStatements = require('../scripts/compare-statements');
+
+const tests = [
+  {
+    input: {
+      __compat: {
+        support: {
+          chrome: [
+            { version_added: '20', prefix: 'webkit' },
+            {
+              version_added: '10',
+              version_removed: '18',
+              partial_implementation: true,
+              notes: 'No fries with the burger',
+            },
+            {
+              version_added: '12',
+              flags: [
+                {
+                  type: 'preference',
+                  name: '#add_fries',
+                  value_to_set: 'Extra Crispy',
+                },
+              ],
+            },
+            { version_added: '20' },
+          ],
+        },
+      },
+    },
+    output: {
+      __compat: {
+        support: {
+          chrome: [
+            { version_added: '20' },
+            {
+              version_added: '10',
+              version_removed: '18',
+              partial_implementation: true,
+              notes: 'No fries with the burger',
+            },
+            { version_added: '20', prefix: 'webkit' },
+            {
+              version_added: '12',
```

Comment:
This order looks a bit complex, why can't the version_added number be the sole standard here?

Reply:
Sometimes, prefixed/alt. named items were implemented _after_ the standard name was implemented.  If we were to only use the `version_added`, we would move those prefixed items to the top, thus making the browser seem like it does not have standard-named support at first glance.

See https://github.com/mdn/browser-compat-data/issues/1596#issuecomment-387552179 (the bottom, expandable part of the comment) for an example.

Reply:
I don't think that's a great example, as expanding the log makes me think version 44 now requires prefix (which is not true).

To me the better way is to:

* 49 - notes: also supports webkit prefixed alias
* 44-49 - notes: also supports webkit prefixed alias / flags: layout.css.prefixes.webkit
* 20
* 18-28 - flags: layout.css.flexbox.enabled

This makes things linear and thus easier to understand. A potentially related example is https://github.com/mdn/browser-compat-data/pull/7387#issuecomment-729484137, where it is linear but does not clearly say that the latest version supports the alias or not.

Reply:
(Or alternatively a subitem to indicate webkit alias support)

Reply:
Looking at the wording, I see what you mean.  However, I don't think that's a problem with the raw BCD data, but rather the display on MDN web docs.  In the case of prefixed support before normal support, it works well since in those earlier versions, the vendor prefix _was_ "required".  Something more like "Supported under the vendor prefix: `[prefix]`" would be more reasonable, perhaps?

Reply:
> Something more like "Supported under the vendor prefix: `[prefix]`" would be more reasonable, perhaps?

That might help, but I think ideally there should be more intuitive way to see it's the "only" support or it's "also" supported. Filed #7403 as this is not exactly a sorting story.